src/pyseekdb/client/sparse_embedding_function.py

```
@@ -0,0 +1,381 @@
+
+        Args:
+            embeddings: Dictionary mapping integer indices to float weights.
+
+        Returns:
+            A new SparseVector instance.
+
+        Example:
+            >>> sv = SparseVector.from_dict({100: 0.5, 200: 0.3, 500: 0.8})
+        """
+        if not isinstance(embeddings, dict):
+            raise TypeError(f"embeddings must be a dict, got {type(embeddings).__name__}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)